### PR TITLE
Plugins Browser: Use react query to search plugins on WPORG

### PR DESCRIPTION
**Pull request by WBerredo**
#### Changes proposed in this Pull Request

* Add useWPORGPlugins to fetch plugins list with react query
* Use the plugin useWPORGPlugins to retrieve data on Plugins Browser

#### Testing instructions
1. Check if the `/plugins` page is working properly (without a selected site)
2. Check if the `/plugins/:site` page is working (with a selected site)
3. Test the search functionality with a keyword for paid and free plugins. ex: `woo` and `yoast`
4. Verify if the pagination is working properly
5. Verify if new API requests are not being done when the same page and same term are requested.

Kapture 2022-02-25 at 16 23 30

### CODE REVIEW

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 1]
```diff
@@ -407,7 +395,16 @@ const SearchListView = ( {
...
+
+	const pluginItemsFeatch = ( currentPage ) => {
+		return currentPage === 1
+			? SEARCH_RESULTS_LIST_LENGTH + ( paidPluginsBySearchTerm?.length % 2 )
+			: SEARCH_RESULTS_LIST_LENGTH;
+	};
+
+	useEffect( () => {
+		setPageSize( pluginItemsFeatch( page ) );
+	}, [ paidPluginsBySearchTerm ] );
```
Comment (gcsecsey) [inline client/my-sites/plugins/plugins-browser/index.jsx]: Small typo here:
```suggestion
	const pluginItemsFetch = ( currentPage ) => {
		return currentPage === 1
			? SEARCH_RESULTS_LIST_LENGTH + ( paidPluginsBySearchTerm?.length % 2 )
			: SEARCH_RESULTS_LIST_LENGTH;
	};

	useEffect( () => {
		setPageSize( pluginItemsFetch( page ) );
	}, [ paidPluginsBySearchTerm ] );
```

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 2]
```diff
@@ -478,8 +458,9 @@ const SearchListView = ( {
 						page={ pluginsPagination.page }
 						perPage={ pluginItemsFeatch( pluginsPagination?.page ) }
 						total={ pluginsPagination.results }
-						pageClick={ ( page ) => {
-							dispatch( fetchPluginsList( null, page, searchTerm, pluginItemsFeatch( page ) ) );
+						pageClick={ ( clickedPage ) => {
+							setPage( clickedPage );
+							setPageSize( pluginItemsFeatch( clickedPage ) );
```
Comment (gcsecsey) [inline client/my-sites/plugins/plugins-browser/index.jsx]: Typo:
```suggestion
							setPageSize( pluginItemsFetch( clickedPage ) );
```

**Review by gcsecsey — APPROVED** [review]
Works great, and LGTM 👍

**Review by epeicher — APPROVED** [review]
Tested and worked as expected. Comparing requests before and after the change return similar results.

LGTM!

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 1 above)
Comment (WBerredo) [reply client/my-sites/plugins/plugins-browser/index.jsx]: Fixed at https://github.com/Automattic/wp-calypso/pull/61501/commits/f0f2d979736885c44ba32ba7b8bfe41793129307

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 2 above)
Comment (WBerredo) [reply client/my-sites/plugins/plugins-browser/index.jsx]: Fixed at https://github.com/Automattic/wp-calypso/pull/61501/commits/f0f2d979736885c44ba32ba7b8bfe41793129307

**Pull request merged**